server/src/main/java/org/elasticsearch/index/mapper/ProvidedIdFieldMapper.java

```diff
@@ -50,9 +50,9 @@
 import java.util.function.Supplier;
 
 /**
- * A mapper for the _id field. It does nothing since _id is neither indexed nor
- * stored, but we need to keep it so that its FieldType can be used to generate
- * queries.
+ * A mapper for the {@code _id} field that that reads the from the
+ * {@link SourceToParse#id()}. It also optionally supports field data
```

Comment:
I feel like `optionally` is redundant here since you already qualify it with `if the cluster...`.